File: app/src/main/java/com/example/android/todolist/mainactivity/MainViewModel.java
package com.example.android.todolist.mainactivity;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;
import android.util.Log;

import com.example.android.todolist.database.AppDatabase;
import com.example.android.todolist.database.TaskEntry;

import java.util.List;

/**
 * Created by raed on 9/20/18.
 */

public class MainViewModel extends AndroidViewModel {
    private static final String TAG = "MainViewModel";

    private LiveData<List<TaskEntry>> tasks;

    public MainViewModel(@NonNull Application application) {
        super(application);
        AppDatabase appDatabase = AppDatabase.getInstance(application.getApplicationContext());
        Log.d(TAG, "MainViewModel: get data from database");
        tasks = appDatabase.taskDao().loadAllTasks();
    }

    public LiveData<List<TaskEntry>> getTasks() {
        return tasks;
    }
}

File: app/src/main/java/com/example/android/todolist/mainactivity/MainViewModel.kt
package com.example.android.todolist.mainactivity

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import com.example.android.todolist.database.AppDatabase
import com.example.android.todolist.database.TaskEntry

/**
 * Created by Raed Saeed on 8/3/2019.
 */
class MainViewModel(application: Application): AndroidViewModel(application) {
    val tasks : LiveData<List<TaskEntry>>

    init {
        val appDatabase : AppDatabase? = AppDatabase.getInstance(application.applicationContext)
        tasks = appDatabase?.taskDao()?.loadAllTasks()!!
    }

    companion object {
        private val TAG = MainViewModel::class.simpleName
    }
}